packages/grafana-data/src/types/dataFrame.ts

```diff
@@ -167,6 +167,12 @@ export interface FieldState {
    */
   seriesIndex?: number;
 
+  /**
+   * Any additional type info that is needed by purpose-specific panels when field name and data type are insufficiant
+   * For instance, an OHLC panel may expect this to be "open_price" to render the top of a candlestick marker
+   */
+  semanticType?: string | null;
+
```

Comment:
Thinking out loud. Maybe instead of introducing this property to a field state we should rely on the semantic mapping config that's passed down to the TimeSeries component anyways? I would rather avoid extending the core type with information that can be transferred by config only.